src/main/java/com/palantir/docker/compose/execution/DefaultDockerCompose.java

```diff
@@ -83,6 +83,36 @@ public void rm() throws IOException, InterruptedException {
     }
 
     @Override
+    public void exec(DockerComposeExecOption dockerComposeExecOption, String containerName,
+            DockerComposeExecArgument dockerComposeExecArgument) throws IOException, InterruptedException {
+        verifyDockerComposeVersion();
+        String[] fullArgs = constructFullDockerComposeExecArguments(dockerComposeExecOption, containerName, dockerComposeExecArgument);
+        executeDockerComposeCommand(throwingOnError(), fullArgs);
+    }
+
+    private void verifyDockerComposeVersion() throws IOException, InterruptedException {
+        String versionOutput = executeDockerComposeCommand(throwingOnError(), "-v");
+        String version = DockerComposeVersions.parseFromDockerComposeVersion(versionOutput);
+        validState(DockerComposeVersions.versionCompare(version, "1.7") >= 0, "You need at least docker-compose 1.7 to run docker-compose exec");
+    }
+
+    private String[] constructFullDockerComposeExecArguments(DockerComposeExecOption dockerComposeExecOption,
```

Comment:
Make the 2 wrapper types return List<String> and then just do:

ImmutableList.builder()
    .add("exec")
    .addAll(options.asList())
    .add(containerName)
    .addAll(arguments.asList())
    .build()

Messing around with arrays is normally a recipe for pain and disaster.